addons/account/models/res_bank.py

```diff
@@ -0,0 +1,95 @@
+# -*- coding: utf-8 -*-
+
+from odoo import api, models, fields, _
+from odoo.exceptions import UserError
+
+class ResPartnerBank(models.Model):
+    _inherit = 'res.partner.bank'
+
+    def build_qr_code_url(self, amount, free_communication, structured_communication, currency, debtor_partner, qr_method=None, silent_errors=True):
+        """ Returns the QR-code report URL to pay to this account with the given parameters,
+        or None if no QR-code could be generated.
+
+        :param amount: The amount to be paid
+        :param free_communication: Free communication to add to the payment when generating one with the QR-code
+        :param structured_communication: Structured communication to add to the payment when generating one with the QR-code
+        :param currency: The currency in which amount is expressed
+        :debtor_partner: The partner to which this QR-code is aimed (so the one who will have to pay)
+        :qr_method: The QR generation method to be used to make the QR-code. If None, the first one giving a result will be used.
+        :silent_errors: If true, forbids errors to be raised if some tested QR-code format can't be generated because of incorrect data.
+        """
+        if not self:
+            return None
```

Comment:
I guess you made this on purpose instead of using a classic `ensure_one`

Reply:
Yup, I don't want it to crash on an empty recordset.